### PR TITLE
Refactor config 

### PR DESCRIPTION
**Pull request by kyscott18**

### CODE REVIEW

File: packages/core/src/config/sources.ts [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+            chainId: network.chainId,
+            events,
+            // optionally overridden properties
+            startBlock: networkContract.startBlock ?? contract.startBlock ?? 0,
+            endBlock: networkContract.endBlock ?? contract.endBlock,
+            maxBlockRange:
+              networkContract.maxBlockRange ?? contract.maxBlockRange,
+          } as const;
+
+          if ("factory" in contract) {
```
Comment (0xOlias) [inline packages/core/src/config/sources.ts:108]: I think this won't "fall through" in the case where `factory` is not defined on the base contract but is defined per-network?

File: packages/core/src/config/sources.ts [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+                ...buildFactoryCriteria(resolvedFactory),
+                topics,
+              },
+            } as const satisfies Factory;
+          } else {
+            // log filter
+
+            const resolvedAddress =
+              ("address" in networkContract && networkContract.address) ||
+              contract.address;
```
Comment (0xOlias) [inline packages/core/src/config/sources.ts:128]: Shouldn't it be possible for `address` to be a list of addresses?

File: packages/core/src/config/sources.ts [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+  >
+): Topics => {
+  if (Array.isArray(filter.event)) {
+    // List of event signatures
+    return [
+      filter.event
+        .map((event) =>
+          encodeEventTopics({
+            abi: [findAbiEvent(abi, event)],
+          })
```
Comment (0xOlias) [inline packages/core/src/config/sources.ts:160]: I wonder if `encodeEventSelector` would be more readable here?

File: packages/core/src/config/sources.ts [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+    // Single event with args
+    return encodeEventTopics({
+      abi: [findAbiEvent(abi, filter.event)],
+      args: filter.args,
+    });
+  }
+};
+
+/**
+ * Finds the abi event for the event string
```
Comment (0xOlias) [inline packages/core/src/config/sources.ts:174]: Finds the event ABI item for the safe event name.

File: packages/core/src/event-store/postgres/migrations.ts [hunk 5]
```diff
@@ -349,6 +349,31 @@ const migrations: Record<string, Migration> = {
         .execute();
     },
   },
+  ["2023_11_06_0_new_rpc_cache_design"]: {
+    async up(db: Kysely<any>) {
+      await db.schema.dropTable("contractReadResults").execute();
+
+      /**
+       * request format
```
Comment (0xOlias) [inline packages/core/src/event-store/postgres/migrations.ts:357]: Formatting for "request" field values:

**Review by 0xOlias — APPROVED** [review]

File: packages/core/src/config/sources.ts (hunk 2 above)
Comment (kyscott18) [reply packages/core/src/config/sources.ts:128]: Yes

File: packages/core/src/config/sources.ts (hunk 1 above)
Comment (kyscott18) [reply packages/core/src/config/sources.ts:108]: Good catch

**Review by 0xOlias — APPROVED** [review]
LGTM

**Pull request merged**